apps/scratch/src/utilities/scratch-student-activities/pipeline.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  CreatePipelineParams,
+  DeletePipelineParams,
+  MovePipelineParams,
+  StudentAction,
+} from "../../types/scratch-student-activities";
+
+import {
+  trackCreateActivity,
+  trackMoveActivity,
+  trackDeleteActivity,
+} from "./index";
+
+type StudentActivityPipelineParams =
+  | CreatePipelineParams
+  | MovePipelineParams
+  | DeletePipelineParams;
+
+export const processStudentActivityPipeline = (
```

Comment:
You should be able to get rid of this function completely. No need for another intermediate step here. This allows you to get rid of all the `pipeline` types and functions and `action` types.

Simply do this logic within `handleStudentActivityTracking`